### PR TITLE
Re-organizing the sections

### PR DESCRIPTION
**Pull request by BasileiosKal**
Proposing the following structure for the document. The main change is that `Sign`, `Verify`, `ProofGen` and `ProofVerify` are moved to their own "core operations" section. This will allow us to more easily define "higher level functions" on top of the "core operations" if need be.

```
|--- Introduction
|      |--- Terminology
|      |--- Notation
|      +--- Organization of this document
|--- Conventions
|--- Considerations
|      |--- Subgroup Selection
|      |--- Messages and Generators
|      +--- Encoding of elements to be hashed
|--- BBS Scheme Definition
|      |--- Parameters
|      |--- Key Generation
|      |      |--- KeyGen
|      |      |--- SkToPk
|      |      +--- KeyValidate        
|      +--- Core Operations
|             |--- Sign
|             |--- Verify
|             |--- ProofGen
|             +--- ProofVerify
|--- Utility Functions
|      |--- CreateGenerators
|      |--- MapMessageToScalar
|      |      +--- MapMessageToScalarAsHash
|      +--- Hash to scalar
|--- Serialization
|      |--- OctetsToSignatures
|      +--- SignatureToOctets
|--- Security Considerations
|--- Ciphersuites 
+--- Appendix
```

### CODE REVIEW

**Comment by tplooker** [comment]
Suggested tweaks

```
|--- Introduction
|      |--- Terminology
|      |--- Notation
|      +--- Organization of this document
|--- Conventions
|--- Considerations
|      |--- Subgroup Selection
|      |--- Messages and Generators
|      +--- Encoding of elements to be hashed
|--- Scheme Definition
|      |--- Parameters
|      |--- Key Generation Operations
|      |      |--- KeyGen
|      |      |--- SkToPk
|      |      +--- KeyValidate        
|      +--- Core Operations
|             |--- Sign
|             |--- Verify
|             |--- ProofGen
|             +--- ProofVerify
|--- Utility Operations
|      |--- CreateGenerators
|      |--- MapMessageToScalar
|      |      +--- MapMessageToScalarAsHash
|      +--- Hash to scalar
|      |--- OctetsToSignatures
|      +--- SignatureToOctets
|--- Security Considerations
|--- Ciphersuite
```

**Comment by BasileiosKal** [comment]
> Suggested tweaks

Done! Thanks!

**Review by tplooker — APPROVED** [review]

**Review by alessandroguggino — APPROVED** [review]

File: draft-bbs-signatures.md [hunk 1]
```diff
@@ -220,47 +220,35 @@ subgroup\_check(P) -> VALID or INVALID
...
 
 Aside from the message generators, the scheme uses two additional generators: `H_s` and `H_d`. The first (`H_s`), is used for the blinding value (`s`) of the signature. The second generator (`H_d`), is used to sign the signature's domain, which binds both signature and proof to a specific context and cryptographically protects any potential application-specific information (for example, messages that must always be disclosed etc.).
 
-### Encoding of elements to be hashed.
+## Encoding of elements to be hashed.
```
Comment (alessandroguggino) [inline draft-bbs-signatures.md:251]: Remove the dot at the end.
```suggestion
## Encoding of elements to be hashed
```

**Comment by tplooker** [comment]
@BasileiosKal with the merge of #155 can you rebase this PR as I think otherwise the section will end up in the wrong place, after that im happy to merge.

**Comment by tplooker** [comment]
@BasileiosKal with the merge of #155 can you rebase this PR as I think otherwise the section will end up in the wrong place, after that im happy to merge.

**Comment by BasileiosKal** [comment]
> @BasileiosKal with the merge of https://github.com/decentralized-identity/bbs-signature/pull/155 can you rebase this PR as I think otherwise the section will end up in the wrong place, after that im happy to merge.

Done rebase! 

Since we have 4 "serialization" operations maybe they are "worth" their own sub-section?? So the `Utility Operations` section now looks like:

```
|--- Utility Operations
|      |--- CreateGenerators
|      |--- MapMessageToScalar
|      |      +--- MapMessageToScalarAsHash
|      +--- Hash to scalar
|      +---- Serialization
|              |--- OctetsToSignatures
|              |--- SignatureToOctets
|              |--- OctetsToProof
|              +--- ProofToOctets
```

I'm more than happy to fall back if people don't like this change.

**Comment by alessandroguggino** [comment]
> Since we have 4 "serialization" operations maybe they are "worth" their own sub-section??
You're right! I agree

**Pull request merged**